src/julia.h

```diff
@@ -157,6 +157,7 @@ typedef struct {
     unsigned short how:2;
     unsigned short isshared:1;  // data is shared by multiple Arrays
     unsigned short isaligned:1; // data allocated with memalign
+    unsigned short fixed_len:1; // data cannot be resized (1d only)
```

Comment:
Might want to rename the field to `resizeable` then since it will no longer actually mean shared. The error message can say that the array either has shared data or came from a map or comprehension.